plugins/slipping-away-users/frontend/public/javascripts/countly.views.js

```diff
@@ -104,18 +110,174 @@ window.slippingView = countlyView.extend({
+                var subs = {};
+                for (var i in filter) {
+                    inputs.push(i);
+                    subs[i] = [];
+                    for (var j in filter[i]) {
+                        if (filter[i][j].length) {
+                            for (var k = 0; k < filter[i][j].length; k++) {
+                                subs[i].push([j, filter[i][j][k]]);
+                            }
+                        }
+                        else {
+                            subs[i].push([j, filter[i][j]]);
+                        }
+                    }
+                }
+                self.setInput(inputs, subs, 0, 0, 1);
+            }
+        }, 500);
+    },
+    setInput: function(inputs, subs, cur, sub, total) {
+        var self = this;
+        sub = sub || 0;
+        if (inputs[cur]) {
+            var filterType = subs[inputs[cur]][sub][0];
+
+            if (filterType === "$in") {
+                filterType = "=";
+            }
+            else if (filterType === "$nin") {
+                filterType = "!=";
+            }
+            var val = subs[inputs[cur]][sub][1];
+            var el = $(".query:nth-child(" + (total) + ")");
+            $(el).data("query_value", val + ""); //saves value as attribute for selected query
+            el.find(".filter-name").trigger("click");
+            el.find(".filter-type").trigger("click");
+
+
+            if (inputs[cur].indexOf("chr.") === 0) {
+                el.find(".filter-name").find(".select-items .item[data-value='chr']").trigger("click");
+                if (val === "t") {
+                    el.find(".filter-type").find(".select-items .item[data-value='=']").trigger("click");
+                }
+                else {
+                    el.find(".filter-type").find(".select-items .item[data-value='!=']").trigger("click");
+                }
+                val = inputs[cur].split(".")[1];
+                subs[inputs[cur]] = ["true"];
+            }
+            else if (inputs[cur] === "did" || inputs[cur] === "chr" || inputs[cur].indexOf(".") > -1) {
+                el.find(".filter-name").find(".select-items .item[data-value='" + inputs[cur] + "']").trigger("click");
+            }
+            else {
+                el.find(".filter-name").find(".select-items .item[data-value='up." + inputs[cur] + "']").trigger("click");
+            }
+
+            el.find(".filter-type").find(".select-items .item[data-value='" + filterType + "']").trigger("click");
+            setTimeout(function() {
+                el.find(".filter-value").not(".hidden").trigger("click");
+                if (el.find(".filter-value").not(".hidden").find(".select-items .item[data-value='" + val + "']").length) {
+                    el.find(".filter-value").not(".hidden").find(".select-items .item[data-value='" + val + "']").trigger("click");
+                }
+                else if (_.isNumber(val) && (val + "").length === 10) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['_' is not defined.](https://app.codacy.com/manual/ar2rsawseen/countly-server/pullRequest?prid=4780185)